src/lib/EventPublisher.js

```diff
@@ -156,11 +164,20 @@ class EventPublisher extends EventEmitter {
         },
         subscriptionTimeout,
       );
-
-      this.once(eventNamePrefix, (): void => clearTimeout(timeout));
+      this.once(eventNamePrefix, () => {
+        cleared = true;
+        clearTimeout(timeout);
+      });
     }
 
     if (once) {
+      if (onceUnsubscribe === true) {
```

Comment:
just instead `nullthrows(this._subscriptionsByID.get(subscriptionID));` in `unusbsribe` do 
```
const subscripiton = this._subscriptionsByID.get(subscriptionID)
 if(!subscripiton) {
  return;
 }
}
```